gnovm/pkg/gnolang/gno_test.go

```diff
@@ -39,6 +39,69 @@ func main() {
 	m.RunMain()
 }
 
+func TestDoOpEvalBaseConversion(t *testing.T) {
+	m := NewMachine("test", nil)
+
+	type testCase struct {
+		input     string
+		expect    string
+		expectErr bool
+	}
+
+	testCases := []testCase{
+		// binary
+		{input: "0b101010", expect: "42", expectErr: false},
+		{input: "0B101010", expect: "42", expectErr: false},
+		{input: "0b111111111111111111111111111111111111111111111111111111111111111", expect: "9223372036854775807", expectErr: false},
+		{input: "0b0", expect: "0", expectErr: false},
+		{input: "0b000000101010", expect: "42", expectErr: false},
+		{input: " 0b101010", expectErr: true},
+		{input: "0b", expectErr: true},
+		{input: "0bXXXX", expectErr: true},
+		{input: "42b0", expectErr: true},
+		// octal
+		{input: "0o42", expect: "34", expectErr: false},
+		{input: "0o0", expect: "0", expectErr: false},
+		{input: "042", expect: "34", expectErr: false},
+		{input: "0777", expect: "511", expectErr: false},
+		{input: "0O0000042", expect: "34", expectErr: false},
+		{input: "0777777777777777777777", expect: "9223372036854775807", expectErr: false},
+		{input: "0o777777777777777777777", expect: "9223372036854775807", expectErr: false},
+		{input: "048", expectErr: true},
+		{input: "0o", expectErr: true},
+		{input: "0oXXXX", expectErr: true},
+		{input: "0OXXXX", expectErr: true},
+		{input: "0x2a ", expectErr: true},
+		// hex
+		{input: "0x2a", expect: "42", expectErr: false},
+		{input: "0X2A", expect: "42", expectErr: false},
+		{input: "0x7FFFFFFFFFFFFFFF", expect: "9223372036854775807", expectErr: false},
+		{input: "0x", expectErr: true},
+		{input: "0xXXXX", expectErr: true},
+		{input: "0xGHIJ", expectErr: true},
+		// decimal
+		{input: "42", expect: "42", expectErr: false},
+		{input: "0", expect: "0", expectErr: false},
+		{input: "0000000000", expect: "0", expectErr: false},
+		{input: "9223372036854775807", expect: "9223372036854775807", expectErr: false},
```

Comment:
Not sure if I missed them, but please add:
- 0420x42
- 0x42o42
- 0o42x42
- 0O42x42

Reply:
Added. thanks